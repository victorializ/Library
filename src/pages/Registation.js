import React, { useState } from 'react';
import { Segment } from 'semantic-ui-react';

import { colors } from "../assets/semantic-colors";
import { register } from '../services/http-client';
import { useRequest } from '../hooks';
import { LoadingComponent, RegistrationForm , RegistrationResult } from '../components';

function Registration() {
  const [ user, setUser ] = useState(false);
  const response = useRequest(register, [user], user);

  return (
    <div className="center-vertical">
      <Segment 
        color={colors.primary} 
        className='registration__form'
      >
        {
          user ? 
            <LoadingComponent {...response}>
              <RegistrationResult data={response.data} />
            </LoadingComponent>
          :
            <RegistrationForm 
              register={user => setUser(user)}
            />
        }
      </Segment>
    </div>
  );
}

export { Registration };
