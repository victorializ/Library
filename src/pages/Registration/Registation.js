import React, { useState } from 'react';

function Registration() {
  const [ email, setEmail ] = useState('');
  const [ password, setPassword ] = useState('');
  const [ firstName, setFirstName ] = useState('');
  const [ lastName, setLastName ] = useState('');
  const [ error, setError ] = useState('');
  const [ checked, check ] = useState(false);

  return (
    <div>Reg</div>
  );
}

export { Registration };