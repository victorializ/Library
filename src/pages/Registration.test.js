import React from 'react';
import { shallow } from 'enzyme';

import { Registration } from './Registation';

describe('Registration page component', () => {
    it('renders correctly', () => {
        expect(shallow(<Registration />)).toMatchSnapshot();
    }); 
});