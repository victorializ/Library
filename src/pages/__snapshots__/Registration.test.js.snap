// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page component renders correctly 1`] = `
<div
  className="center-vertical"
>
  <Segment
    className="registration__form"
    color="olive"
  >
    <RegistrationForm
      register={[Function]}
    />
  </Segment>
</div>
`;
